torchgen/model.py

```diff
@@ -886,6 +908,7 @@ def __post_init__(self) -> None:
     def has_composite_kernel(self) -> bool:
         return (
             self.has_composite_implicit_autograd_kernel
+            or self.has_composite_implicit_autograd_nested_tensor_kernel
```

Comment:
Is this function actually used anywhere in the codegen?

I just grepped around and I don't think it is. If that's the case, do you mind deleting it in this PR?

It would be a pretty confusing function to use now that we have `CompositeImplicitAutogradNestedTensor`: you might have a function that returns `f.has_composite_kernel() == True`, but it only has a decomposition that works for nested tensors, and not for normal tensors, which probably isn't what you expect.

Reply:
Tried deleting it but got errors. It should be used in some way.

Changed to
```
    @property
    def has_composite_kernel(self) -> bool:
        return (
            self.has_composite_implicit_autograd_kernel
            or self.has_composite_explicit_autograd_kernel
            or self.has_composite_explicit_autograd_non_functional_kernel
        ) or (
            self.has_composite_implicit_autograd_kernel
            and self.has_composite_implicit_autograd_nested_tensor_kernel
        )
```